src/Common/src/CoreLib/Interop/Unix/System.Native/Interop.GetTimestamp.cs

```diff
@@ -8,10 +8,10 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetTimestampResolution")]
-        internal static extern bool GetTimestampResolution(out long resolution);
+        [DllImport(Libraries.SystemNative, BestFitMapping = false, CallingConvention = CallingConvention.Cdecl, EntryPoint = "SystemNative_GetTimestampResolution", ExactSpelling = true, PreserveSig = true, SetLastError = false, ThrowOnUnmappableChar = false)]
+        internal static extern unsafe int GetTimestampResolution(long* resolution);
 
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetTimestamp")]
-        internal static extern bool GetTimestamp(out long timestamp);
+        [DllImport(Libraries.SystemNative, BestFitMapping = false, CallingConvention = CallingConvention.Cdecl, EntryPoint = "SystemNative_GetTimestamp", ExactSpelling = true, PreserveSig = true, SetLastError = false, ThrowOnUnmappableChar = false)]
+        internal static extern unsafe int GetTimestamp(long* timestamp);
```

Comment:
Using bool instead of int forces measurable additional marshaling overhead?

Reply:
Yup, because `bool` is not blittable. The int has to be both marshalled and normalized.

Reply:
Also, bool<->int32_t marshaling is Windowsism that feels foreign in the Unix world. We have a few outliers in Unix shims that depend on bool marshaling, but majority of our Unix shims PInvokes just use `int` even if they could have used `bool`.

I think changing this to return `int` is fine.

Reply:
I am wondering whether it is worth it to return error here at all. I do not see any places where we check the error - we expect this method to always succeed.

Reply:
> Also, bool<->int32_t marshaling is Windowsism that feels foreign in the Unix world. We have a few outliers in Unix shims that depend on bool marshaling, but majority of our Unix shims PInvokes just use int even if they could have used bool. I think changing this to return int is fine.

My preference then would be to systematically change all such usage in all of our P/Invokes.  Doesn't need to block this PR, but it'd be good to proactively get there.

Reply:
> I am wondering whether it is worth it to return error here at all.

We don't set `GetLastError=true` and so that should avoid any overhead that would exist. We do check the result in the case of `QueryPerformanceFrequency` (which determines whether we are using DateTime or `QueryPerformanceCounter` for time measurements). We never consume the result in the case of `QueryPerformanceCounter`, but there should be no downside to having the P/Invoke signature match here.